### DISCUSSION TITLE
workflow: add-apt-repository failing for ppa:maxmind/ppa

### DISCUSSION THREAD

**Discussion by martinhsv** [community/community#69720] [Packages] [answered]
### Select Topic Area

Bug

### Body

A project I manage ( https://github.com/SpiderLabs/ModSecurity ) has begun to have a workflow failure that I have not seen before.

The following command results in an error:
```
sudo add-apt-repository --yes ppa:maxmind/ppa
```

-----
2023-10-10T18:03:38.1001270Z Cannot add PPA: 'ppa:~maxmind/ubuntu/ppa'.
2023-10-10T18:03:38.1001943Z The team named '~maxmind' has no PPA named 'ubuntu/ppa'
2023-10-10T18:03:38.1002376Z Please choose from the following available PPAs:
2023-10-10T18:03:38.1002851Z  * 'ppa':  MaxMind Libraries and Software
-----

This same workflow succeeded on Sept. 29.

Also, I am able to run the command successfully on one of my own Ubuntu 20.04 machines, so it doesn't seem likely that the repository itself has a problem.

Is there some known change at github that would cause this?

**Comment by akien-mga** [comment]
Same issue for Godot with https://launchpad.net/~kisak/+archive/ubuntu/kisak-mesa

![image](https://github.com/community/community/assets/4701338/0008887b-245f-4c23-8048-23ee705a2099)

It started failing sporadically 2 or 3 days ago. Restarting the jobs can make it pass, it's not clear what causes it to fail in the first place.

**Reply by akien-mga** [reply]
We've implemented a workaround to add the ppa sources manually instead of using the seemingly broken `add-apt-repository`. So far it seems to work: https://github.com/godotengine/godot/pull/83214

**Comment by sebhofmann** [comment]
We have the same issue. Another workflow run of the PR just works.
<img width="907" alt="Bildschirmfoto 2023-10-13 um 13 11 07" src="https://github.com/community/community/assets/7668803/2dc31446-2d19-44ef-a593-43abfc88cba1">

**Comment by martinhsv** [comment]
I wound up working around this issue by advancing the os to a newer version. With the more recent os version, the regular repo has a sufficiently high version of the software I need such that I was able to remove the add-apt-repository.

Of course, this solution won't work for everyone.

**Comment by cjwatson** [comment]
I'm working on this in https://bugs.launchpad.net/launchpad/+bug/2039251.

**Comment by cjwatson** [comment, marked as answer]
This should now be fixed so that workarounds aren't necessary.
